### ISSUE TITLE
cFuzzySet usage: all match scores are 0?

### ISSUE THREAD

**Issue by nodice73** [axiak/fuzzyset#21]
Hello,

When I use cFuzzySet instead of FuzzySet, every match returns a list of the same length, with all zero similarity scores. I tried changing the rel_sim_cutoff parameter, but it doesn't appear to do anything. Is this the intended behavior?

FuzzySet returns matches with scores, and the match lists are different lengths depending on the scores, which is what I expected.

Thank you!

**Comment by axiak** [comment]
Can you provide a code sample?

**Comment by nodice73** [comment]
Using cfuzzyset:

```
from cfuzzyset import cFuzzySet as FuzzySet

uptags = ['ACTATATGTGAAGGCATGGC', 'AGTCCATGCAGCACTCTTGG',
             'AACAGTTCCGAAGGCTGAGC', 'AAAGGAATCTGTCTCAACGC',
             'ACATCGAGGGAATACGACGC', 'TAGTATCGGAACTCGCCGC',
             'ATTCGATGGGAACGATCCGC', 'ATACTCTCGGAAGGTTGCGC',
             'ACGTATAGCATAAGGTAGGC']

barcodess = ['TACCTAGCGCGAGGACTGAT',  'GGGTCCAGTATAGTCGATAT', 
                   'CAGACTTGGAAACCTTCTCA', 'AGGAGTCAAGTGTGTA',
                   'CATCGTTATGGCAGGGATAT', 'AAGTCGAGATAATCGCCTC',
                   'GAGTTACTACCAACCATTAG', 'ATAGGCGGACCACCTCTAA',
                   'ACCTCGCTTGAATGACG']

fs = FuzzySet(iterable=uptags)
for barcode in barcodess:
    matches = fs.get(barcode)
    print(matches)

#### Output:
[(0.0, 'ACATCGAGGGAATACGACGC'), (0.0, 'ATACTCTCGGAAGGTTGCGC'), (0.0, 'AACAGTTCCGAAGGCTGAGC'), (0.0, 'TAGTATCGGAACTCGCCGC'), (0.0, 'ATTCGATGGGAACGATCCGC'), (0.0, 'ACGTATAGCATAAGGTAGGC'), (0.0, 'AAAGGAATCTGTCTCAACGC'), (0.0, 'ACTATATGTGAAGGCATGGC'), (0.0, 'AGTCCATGCAGCACTCTTGG')]
[(0.0, 'ACGTATAGCATAAGGTAGGC'), (0.0, 'TAGTATCGGAACTCGCCGC'), (0.0, 'AGTCCATGCAGCACTCTTGG'), (0.0, 'ATTCGATGGGAACGATCCGC'), (0.0, 'ACATCGAGGGAATACGACGC'), (0.0, 'ACTATATGTGAAGGCATGGC'), (0.0, 'AACAGTTCCGAAGGCTGAGC'), (0.0, 'ATACTCTCGGAAGGTTGCGC'), (0.0, 'AAAGGAATCTGTCTCAACGC')]
[(0.0, 'AGTCCATGCAGCACTCTTGG'), (0.0, 'AAAGGAATCTGTCTCAACGC'), (0.0, 'ATACTCTCGGAAGGTTGCGC'), (0.0, 'TAGTATCGGAACTCGCCGC'), (0.0, 'ATTCGATGGGAACGATCCGC'), (0.0, 'AACAGTTCCGAAGGCTGAGC'), (0.0, 'ACATCGAGGGAATACGACGC'), (0.0, 'ACTATATGTGAAGGCATGGC')]
[(0.0, 'AAAGGAATCTGTCTCAACGC'), (0.0, 'ACTATATGTGAAGGCATGGC'), (0.0, 'AACAGTTCCGAAGGCTGAGC'), (0.0, 'ACGTATAGCATAAGGTAGGC'), (0.0, 'AGTCCATGCAGCACTCTTGG'), (0.0, 'TAGTATCGGAACTCGCCGC'), (0.0, 'ATACTCTCGGAAGGTTGCGC'), (0.0, 'ACATCGAGGGAATACGACGC'), (0.0, 'ATTCGATGGGAACGATCCGC')]
[(0.0, 'ACTATATGTGAAGGCATGGC'), (0.0, 'ACGTATAGCATAAGGTAGGC'), (0.0, 'ATTCGATGGGAACGATCCGC'), (0.0, 'ACATCGAGGGAATACGACGC'), (0.0, 'AGTCCATGCAGCACTCTTGG'), (0.0, 'TAGTATCGGAACTCGCCGC'), (0.0, 'ATACTCTCGGAAGGTTGCGC'), (0.0, 'AACAGTTCCGAAGGCTGAGC'), (0.0, 'AAAGGAATCTGTCTCAACGC')]
[(0.0, 'TAGTATCGGAACTCGCCGC'), (0.0, 'ACATCGAGGGAATACGACGC'), (0.0, 'ATTCGATGGGAACGATCCGC'), (0.0, 'AAAGGAATCTGTCTCAACGC'), (0.0, 'ATACTCTCGGAAGGTTGCGC'), (0.0, 'AACAGTTCCGAAGGCTGAGC'), (0.0, 'ACGTATAGCATAAGGTAGGC'), (0.0, 'AGTCCATGCAGCACTCTTGG'), (0.0, 'ACTATATGTGAAGGCATGGC')]
[(0.0, 'AGTCCATGCAGCACTCTTGG'), (0.0, 'AACAGTTCCGAAGGCTGAGC'), (0.0, 'ATACTCTCGGAAGGTTGCGC'), (0.0, 'TAGTATCGGAACTCGCCGC'), (0.0, 'ACATCGAGGGAATACGACGC'), (0.0, 'ACTATATGTGAAGGCATGGC'), (0.0, 'ACGTATAGCATAAGGTAGGC'), (0.0, 'AAAGGAATCTGTCTCAACGC'), (0.0, 'ATTCGATGGGAACGATCCGC')]
[(0.0, 'ATACTCTCGGAAGGTTGCGC'), (0.0, 'ACGTATAGCATAAGGTAGGC'), (0.0, 'AAAGGAATCTGTCTCAACGC'), (0.0, 'ACTATATGTGAAGGCATGGC'), (0.0, 'AGTCCATGCAGCACTCTTGG'), (0.0, 'TAGTATCGGAACTCGCCGC'), (0.0, 'ACATCGAGGGAATACGACGC'), (0.0, 'AACAGTTCCGAAGGCTGAGC'), (0.0, 'ATTCGATGGGAACGATCCGC')]
[(0.0, 'ACATCGAGGGAATACGACGC'), (0.0, 'ATACTCTCGGAAGGTTGCGC'), (0.0, 'TAGTATCGGAACTCGCCGC'), (0.0, 'ACTATATGTGAAGGCATGGC'), (0.0, 'AAAGGAATCTGTCTCAACGC'), (0.0, 'ATTCGATGGGAACGATCCGC'), (0.0, 'AACAGTTCCGAAGGCTGAGC'), (0.0, 'AGTCCATGCAGCACTCTTGG'), (0.0, 'ACGTATAGCATAAGGTAGGC')]
```

Using fuzzyset:

```
from fuzzyset import FuzzySet as FuzzySet

fs = FuzzySet(iterable=uptags)
for barcode in barcodess:
    matches = fs.get(barcode)
    print(matches)

##### Output
[(0.5, 'AACAGTTCCGAAGGCTGAGC')]
[(0.44999999999999996, 'AGTCCATGCAGCACTCTTGG')]
[(0.5, 'ATACTCTCGGAAGGTTGCGC'), (0.5, 'TAGTATCGGAACTCGCCGC')]
[(0.5, 'AACAGTTCCGAAGGCTGAGC')]
[(0.5, 'ACTATATGTGAAGGCATGGC')]
[(0.65, 'ACATCGAGGGAATACGACGC')]
[(0.44999999999999996, 'AGTCCATGCAGCACTCTTGG')]
[(0.44999999999999996, 'AGTCCATGCAGCACTCTTGG')]
[(0.65, 'ACATCGAGGGAATACGACGC')]

**Comment by axiak** [comment]
this seems to be a disparity in how levenshtein is working in the cython version. I'll take a look. In the meantime it might make sense to disable levenshtein for your task anyway.

**Comment by axiak** [comment]
Found the bug. will push a fix.

**Comment by axiak** [comment]
I just published 0.0.18 -- can you check that it works as you'd expect?

**Comment by axiak** [comment]
nvm -- still working on it

**Comment by axiak** [comment]
OK this is fixed in `0.0.19`.

**Comment by nodice73** [comment]
Thank you for the quick fix! It does work now.

Can you explain why I should not use Levenshtein distances for my case? Sorry, I am new to this.

Thank you.
